pkg/worker/worker.go

```diff
@@ -34,7 +34,11 @@ func (w *Worker) Run(work chan events.Event, stopChannel chan struct{}) {
 	for {
 		select {
 		case event := <-work:
-			if shouldProcess, reason := w.ShouldProcess(event); !shouldProcess {
+			if err := w.MutateAKS(event); err != nil {
```

Comment:
`MutateAKS` can be called after `ShouldProcess`. ShouldProcess skips events which are of no interest to the ingress controller as whole.

Reply:
`ShouldProcess` is now renamed to `ShouldMutateAppGateway`.
I'm preserving the functionality of `ShouldMutateAppGateway`, which I find to be strictly related to App Gateway stuff.

`MutateAKS` will make its own determinations on whether to mutate AKS or not.

Reply:
Actually - let me revert that back.  We can simplify this at a later stage - not to boil the ocean in one PR.